sympy/matrices/matrices.py

```diff
@@ -1053,9 +1053,26 @@ def extract(self, rowsList, colsList):
         [2],
         [8]])
 
+        when colsList or rowsList is boolean then indexing will start at 0.
+
+        >>> m.extract([0, 1, 2, 3], [True, False, True])
+        Matrix([
+        [0,  2],
+        [3,  5],
+        [6,  8],
+        [9, 11]])
         """
+
         cols = self.cols
         flat_list = self._mat
+        if rowsList and all(isinstance(i, bool) for i in rowsList):
+            rowsList = [index for index, item in enumerate(rowsList) if item]
+            if not rowsList:
+                return
```

Comment:
No, that's not return empty matrix when we use boolean value of rowList or colsList. if we need that then i will remove 
 ```
if not rowsList:
      return
```
that condition then it's return empty Matrix.